### PR TITLE
NWB, CI and file splitting

### PR DESCRIPTION
**Pull request by MaximilianPichler**
Hey, 
I've made a new branch since I made a mistake with the old one. (was not mergeable)
One sidenote: Please enable Travis beforehand.

### CODE REVIEW

**Review by TarikHuber — APPROVED** [review]
OK

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
+  Utilities functions and polyfills
```
Comment (FDiskas) [inline src/utils.js:2]: Useless file

**Comment by FDiskas** [comment]
I like this refactor. There is still missing some stuff like jsLint, CodingStyle rules and [prehooks](https://www.npmjs.com/package/husky) :christmas_tree:

File: src/utils.js (hunk 1 above)
Comment (Sharlaan) [reply src/utils.js:2]: those exported functions are critical ...

Comment (FDiskas) [reply src/utils.js:2]: Check my pull request - they are not needed

**Comment by FDiskas** [comment]
What?
This crap got merged? :)
did you noticed that:
1. package.json "main" section is wrong. 
2. The NWB builds component in wrong format that IDE like WebStorm is not capable to autosuggestes props for component
3. the .npmignore file with a wrong content. So the "demo" directory will be published in to npm
4. Utils.js is a mess and usles file - and he will be published to npm to, but used only in tests?
5. You added a eslint but source code is not following the rules.
6. There is allot npm peer dependency issues.

**Comment by TarikHuber** [comment]
@FDiskas than make it better

**Comment by FDiskas** [comment]
Tried, but got rejected :)

**Comment by TarikHuber** [comment]
You can sen PR-s to fix the bugs you mentioned. If they fix the problems they will be merged.

**Comment by FDiskas** [comment]
:barber: @TarikHuber you can always grab the ready to go boiler plate https://github.com/nfq-eta/nfq-react-component

**Comment by TarikHuber** [comment]
Thx for the link and the boilerplate. But that is not enough for us to use it here. There is no docu and not enough collaborators to use it safely. We need something that will be updated with time and is maintained by a larger community to avoid abandoned boilerplate's. You can play with the configs of NWB and set everything up for your needs. As long as hey don't break the current state they will be merged. Sending us a boilerplate is nice but not enough. You have to put some work in it to have it as you want.

**Comment by FDiskas** [comment]
So I just can suggest you the folowing:
1. Change your coding style to https://github.com/airbnb/javascript
2. Write tests to cover 100% of your code.
3. Do not allow to merge code if tests | cover | lint | e2e fails
4. Demo should be with a documentation. e.x. use "storrybook"
5. Automate publishing to npm with travis
6. Write CONTRIBUTING.md and other templates https://github.com/blog/2111-issue-and-pull-request-templates
7. Create some "public chat" e.x. using https://gitter.im (so it could work as support or discuss about some ideas)
8. Use https://waffle.io or similar to handle issues and TODOS in better way.
9. And an old school way - create a mailing list - to get ability to inform all contributors about some important stuff.
10. Add some security gate using https://snyk.io
11. Add code quality gate using https://deepscan.io

Good luck :+1:

**Comment by TarikHuber** [comment]
Thx @FDiskas for the list. Some of those are already implemented like nr. 5, 3. We are working on others like nr. 4 and for the rest we need more time and contributors. You are welcome to join us 😄

**Comment by Sharlaan** [comment]
Thanks for your suggestions, here are my answers :

1. i always followed Standard code style, not airbnb. eslint-config-jonnybuchanan is extends Stanrd and Standard-React.
2. tests are planned, be patient...
3. this is already implemented through Travis, CodeCov and Deepscan + linting locally in IDE
4. Maximilian is on it this week, be patient....
5. Have you actually look in the content of the merge ?
6. there is a squeleton thanks to NWB, will be edited further later...
7. and 8. This a simple lib, not a full framework like Material-UI XD. Won't bother, Issues section is more than enough.
9. It's your duty to check for updates, authors never played with subscription/publish system for their open source products. That's for paid's, which noone would want right ?
10. It is also your duty to check for security. By the way, a selectfield component will never be used to request server directly. If security had to be verified, would be on the DAO layer.
11. Deepscan is already used since 2 months ...

**Comment by Sharlaan** [comment]
> 1. package.json "main" section is wrong.

As Tarik explained, please have a look at NWB documentation about "main" and "module" builds.

> 2. The NWB builds component in wrong format that IDE like WebStorm is not capable to autosuggestes props for component

Can you develop an example ? i never heard/saw about props autocompletion, interesting.

> 3. the .npmignore file with a wrong content. So the "demo" directory will be published in to npm

Good point. Thanks.

> 4. Utils.js is a mess and useless file - and he will be published to npm to, but used only in tests?

We splitted the monolithic original SuperSelectField.js into several files. Utils.js gets the utilities functions used by the lib, i remember to have explained this twice not so long ago ...

> 5. You added a eslint but source code is not following the rules.

Which rules are missing ? Please take habit to back your sayings with examples.

> 6. There is a lot npm peer dependency issues.

again, please post a quote.

**Comment by FDiskas** [comment]
2. and 1. Already done and created boiler plate https://github.com/nfq-eta/nfq-react-component
5. Just run `npm run lint` and check console
6. Just run `npm install` and check console

**Comment by Sharlaan** [comment]
For the linting, yea i forgot to run the 1st pass, will be automated on next PRs no worries.

For peer dependencies, just be sure, a peer dep by definition is NOT provided by the component, but from the project context.
So obviously you gonna get issues if you don't install beforehand the required dependencies.
Also, those issues gets resolved as soon as you install the required deps.
In SuperSelectfield case, the codumentation clearly states 3 required peer deps:
- React
- React-dom
- Material-UI
... which you should have installed in the first place anyways ....

**Comment by FDiskas** [comment]
Thanks for the response.
Got one more idea to help you up with your project
Check this out https://probot.github.io/apps/
Here is some blog post about it https://medium.com/ahmed-t-ali/lets-build-a-github-pro-bot-5e155cec395f

**Pull request merged**